internal/internal_task_handlers.go

```diff
@@ -1028,10 +1028,16 @@ ProcessEvents:
 		}
 		if isReplay {
 			w.workflowInfo.currentTaskBuildID = currentBuildID
-		} else if !isReplay && !isQueryOnlyTask {
+		} else if !isQueryOnlyTask {
 			// Query only tasks should use the build ID from the workflow task, not the worker's
 			// build id, since the user cares about what affected workflow state.
 			w.workflowInfo.currentTaskBuildID = w.wth.workerBuildID
+		} else {
+			// It is a query only task, and we're not replaying somehow - still use the task's build
```

Comment:
I   don't  understand this comment, didn't we discuss before that queries are always treated as not replaying?

Reply:
Yeah, I can just get rid of the not replaying part